### PR TITLE
test(java-binding): reduce time of e2e java binding test

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This fixes https://github.com/risingwavelabs/risingwave/issues/9423 and reduce the CI time of e2e java binding test by reusing the java package built from the `build-other` part of the CI and avoiding extra build in the e2e java binding test.

In the latest CI of this PR, the e2e java binding test takes less than 3 minutes to finish, which is much fast, and is unlikely to fail again.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by yufansong — APPROVED** [review]
LGTM

File: ci/scripts/build-other.sh [hunk 1]
```diff
@@ -9,10 +9,22 @@ source ci/scripts/common.sh
...
+mv target/debug/data-chunk-payload-convert-generator bin/data-chunk-payload-convert-generator
+mv target/debug/data-chunk-payload-generator bin/data-chunk-payload-generator
+tar cf java-binding-integration-test.tar bin java/java-binding-integration-test/target/dependency java/java-binding-integration-test/target/classes
+
 echo "--- Upload Java artifacts"
 cp java/connector-node/assembly/target/risingwave-connector-1.0.0.tar.gz ./risingwave-connector.tar.gz
 cp java/udf-example/target/risingwave-udf-example.jar ./risingwave-udf-example.jar
 buildkite-agent artifact upload ./risingwave-connector.tar.gz
 buildkite-agent artifact upload ./risingwave-udf-example.jar
+buildkite-agent artifact upload ./java-binding-integration-test.tar
```
[the remark this answers is not in the document]
Comment (xxchan) [reply ci/scripts/build-other.sh:30]: i.e., `tar --zstd`

[the remark this answers is not in the document]
Comment (wenym1) [reply ci/scripts/build-other.sh]: The uploaded artifact is already a compressed tar. 

Seems that on ubuntu without any extra parameter the tar command does no compression, but on my mac the same command can produces a file with only 20+ MB. 

Just tried specifying the compression algorithms in the tar command.

[the remark this answers is not in the document]
Comment (wenym1) [reply ci/scripts/build-other.sh]: After specifying zstd, the tar size is reduced to 300+ MB.

**Comment by TennyZhuang** [comment]
Can you take a look at the main CI failed?

https://buildkite.com/risingwavelabs/main/builds/4686#0188fcbe-fbc5-4122-aa60-bffd270f3517

**Comment by wenym1** [comment]
> Can you take a look at the main CI failed?

Fixed with https://github.com/risingwavelabs/risingwave/pull/10612

**Pull request merged**